File: java/50 DiagonalSum2DArray.java
package code;
import java.util.*;
public class Test {

	public static void main(String[] args) {
		Scanner sc=new Scanner(System.in);
		int arr1[][]= {{1,2,3},{4,5,6},{7,8,9}};
		int sum=0;
		boolean found=false;
		int n=arr1.length;
		for(int i=0;i<n;i++) {
			sum+=arr1[i][i];
			if(i!=n-1-i) {
				sum+=arr1[i][n-1-i];
			}
		}
		System.out.println("Sum is "+sum);
	}
}

File: python/50 DiagonalSum2DArray.py
arr=[[1,2,3],[4,5,6],[7,8,9]]
sum=0
n=len(arr)
for i in range(n):
    sum+=arr[i][i]
    if i is not n-1-i:
        sum+=arr[i][n-1-i]
print("Sum is ",sum)